src/NuGet.Clients/NuGet.Credentials/PluginCredentialProvider.cs

```diff
@@ -0,0 +1,151 @@
+        /// <param name="isRetry">If true, credentials were previously supplied by this
+        /// provider for the same uri.</param>
+        /// <param name="nonInteractive">If true, the plugin must not prompt for credentials.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        /// <returns>A credential object.  If </returns>
+        public Task<ICredentials> Get(Uri uri, IWebProxy proxy, bool isProxyRequest, bool isRetry,
+            bool nonInteractive, CancellationToken cancellationToken)
+        {
+            if(isProxyRequest)
+            {
+                return Task.FromResult((ICredentials)null);
+            }
+
+            PluginCredentialResponse response = null;
+            try
+            {
+                var request = new PluginCredentialRequest
+                {
+                    Uri = uri.ToString(),
+                    IsRetry = isRetry,
+                    NonInteractive = nonInteractive
+                };
+
+                response = Execute(request, cancellationToken);
+            }
+            catch(PluginException)
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw PluginException.Create(Path, e);
+            }
+
+            if (response.Abort)
+            {
+                throw PluginException.CreateAbortMessage(Path, response.AbortMessage ?? string.Empty);
+            }
+
+            var result = response.IsValid ? new NetworkCredential(response.Username, response.Password) : null;
+            var task = Task.FromResult((ICredentials)result);
+
+            return task;
+        }
+
+        /// <summary>
+        /// Path to plugin credential provider executable.
+        /// Internal for testability.
+        /// </summary>
+        internal string Path { get; }
+
+        /// <summary>
+        /// Seconds to wait for plugin credential service to respond.
+        /// Internal for testability.
+        /// </summary>
+        internal int TimeoutSeconds { get; }
+
+        public virtual PluginCredentialResponse Execute(PluginCredentialRequest request,
+            CancellationToken cancellationToken)
+        { 
+            string requestString = string.Concat(JsonConvert.SerializeObject(request), Environment.NewLine);
+
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = Path,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                UseShellExecute = false,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                ErrorDialog = false
+            };
+
+            var process = Process.Start(startInfo);
+            if (process == null)
+            {
+                throw PluginException.CreateNotStartedMessage(Path);
+            }
+
+            process.StandardInput.WriteLine(requestString);
```

Comment:
use params instead of using the StandardInput, and also make sure to pass encoding